# Guided Capstone Step 4. Pre-Processing and Training Data Development

**The Data Science Method**  


1.   Problem Identification 


2.   Data Wrangling 
  
 
3.   Exploratory Data Analysis   

4.   **Pre-processing and Training Data Development**  
 * Create dummy or indicator features for categorical variables
  * Standardize the magnitude of numeric features
  * Split into testing and training datasets
  * Apply scaler to the testing set
5.   Modeling 
  * Fit Models with Training Data Set
  * Review Model Outcomes — Iterate over additional models as needed.
  * Identify the Final Model

6.   Documentation
  * Review the Results
  * Present and share your findings - storytelling
  * Finalize Code 
  * Finalize Documentation

**<font color='teal'> Start by loading the necessary packages as we did in step 3 and printing out our current working directory just to confirm we are in the correct project directory. </font>**

In [1]:
import os
import pandas as pd
import datetime
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
%matplotlib inline

In [2]:
os.getcwd()

'C:\\Users\\Grace-2019\\Documents\\Data_analyst\\springboard_data_camp\\unit6_Guided_CapstoneProject1'

**<font color='teal'>  Load the csv file you created in step 3, remember it should be saved inside your data subfolder and print the first five rows.</font>**

In [3]:
df=pd.read_csv('step3_output.csv')

In [58]:
df.head()

Unnamed: 0  Unnamed: 0.1                           Name       state  \
0           1             1            Eaglecrest Ski Area      Alaska   
1           2             2               Hilltop Ski Area      Alaska   
2           4             4            Sunrise Park Resort     Arizona   
3           5             5  Yosemite Ski & Snowboard Area  California   
4           8             8         Boreal Mountain Resort  California   

   summit_elev  vertical_drop  trams  fastEight  fastSixes  fastQuads  ...  \
0         2600           1540      0        0.0          0          0  ...   
1         2090            294      0        0.0          0          0  ...   
2        11100           1800      0        0.0          0          1  ...   
3         7800            600      0        0.0          0          0  ...   
4         7700            500      0        0.0          0          1  ...   

   SkiableTerrain_ac  Snow Making_ac  daysOpenLastYear  yearsOpen  \
0              640.0            60.0              45.0       44.0   
1               30.0            30.0             150.0       36.0   
2              800.0            80.0             115.0       49.0   
3               88.0             0.0             110.0       84.0   
4              380.0           200.0             150.0       54.0   

   averageSnowfall  AdultWeekday  AdultWeekend  projectedDaysOpen  \
0            350.0          47.0      53.00000               90.0   
1             69.0          30.0      34.00000              152.0   
2            250.0          74.0      78.00000              104.0   
3            300.0          47.0      47.00000              107.0   
4            400.0          49.0      64.16681              150.0   

   NightSkiing_ac  cluster  
0             0.0        2  
1            30.0        0  
2            80.0        1  
3             0.0        1  
4           200.0        1  

[5 rows x 28 columns]

In [5]:
df.describe().T

count         mean          std    min     25%     50%  \
Unnamed: 0         157.0   167.719745    96.460822    1.0    80.0   171.0   
Unnamed: 0.1       157.0   167.719745    96.460822    1.0    80.0   171.0   
summit_elev        157.0  3714.038217  3226.731258  315.0  1270.0  2025.0   
vertical_drop      157.0   851.738854   574.433212  175.0   400.0   700.0   
trams              157.0     0.000000     0.000000    0.0     0.0     0.0   
fastEight          157.0     0.000000     0.000000    0.0     0.0     0.0   
fastSixes          157.0     0.000000     0.000000    0.0     0.0     0.0   
fastQuads          157.0     0.242038     0.592704    0.0     0.0     0.0   
quad               157.0     0.522293     0.703344    0.0     0.0     0.0   
triple             157.0     1.216561     1.257381    0.0     0.0     1.0   
double             157.0     1.757962     1.503847    0.0     1.0     1.0   
surface            157.0     2.222930     1.421442    0.0     1.0     2.0   
total_chairs       157.0     5.961783     2.770924    0.0     4.0     5.0   
Runs               157.0    29.840764    17.499271    0.0    16.0    25.0   
TerrainParks       157.0     1.808917     1.410282    0.0     1.0     2.0   
LongestRun_mi      157.0     1.052229     0.726564    0.0     0.5     1.0   
SkiableTerrain_ac  157.0   230.738854   281.623866    0.0    60.0   125.0   
Snow Making_ac     157.0    84.738854    72.492453    0.0    30.0    60.0   
daysOpenLastYear   157.0    92.165605    40.115833    0.0    75.0   100.0   
yearsOpen          157.0    58.484076    14.732489   22.0    50.0    58.0   
averageSnowfall    157.0   144.006369   110.810389    0.0    50.0   108.0   
AdultWeekday       157.0    50.425766    15.874440   17.0    40.0    49.0   
AdultWeekend       157.0    57.638274    15.314742   20.0    47.0    58.0   
projectedDaysOpen  157.0   109.063694    26.975697   30.0    93.0   107.0   
NightSkiing_ac     157.0    44.471338    48.790967    0.0     0.0    35.0   
cluster            157.0     0.617834     0.772401    0.0     0.0     0.0   

                           75%      max  
Unnamed: 0          250.000000    327.0  
Unnamed: 0.1        250.000000    327.0  
summit_elev        6550.000000  12075.0  
vertical_drop      1150.000000   2610.0  
trams                 0.000000      0.0  
fastEight             0.000000      0.0  
fastSixes             0.000000      0.0  
fastQuads             0.000000      2.0  
quad                  1.000000      2.0  
triple                2.000000      5.0  
double                2.000000      6.0  
surface               3.000000      6.0  
total_chairs          7.000000     16.0  
Runs                 41.000000     83.0  
TerrainParks          2.000000      6.0  
LongestRun_mi         1.500000      3.0  
SkiableTerrain_ac   240.000000   1500.0  
Snow Making_ac      135.000000    300.0  
daysOpenLastYear    118.000000    175.0  
yearsOpen            67.000000     95.0  
averageSnowfall     210.000000    500.0  
AdultWeekday         57.916957     89.0  
AdultWeekend         68.000000     93.0  
projectedDaysOpen   126.000000    200.0  
NightSkiing_ac       73.000000    200.0  
cluster               1.000000      2.0

In [6]:
df.dtypes

Unnamed: 0             int64
Unnamed: 0.1           int64
Name                  object
state                 object
summit_elev            int64
vertical_drop          int64
trams                  int64
fastEight            float64
fastSixes              int64
fastQuads              int64
quad                   int64
triple                 int64
double                 int64
surface                int64
total_chairs           int64
Runs                 float64
TerrainParks         float64
LongestRun_mi        float64
SkiableTerrain_ac    float64
Snow Making_ac       float64
daysOpenLastYear     float64
yearsOpen            float64
averageSnowfall      float64
AdultWeekday         float64
AdultWeekend         float64
projectedDaysOpen    float64
NightSkiing_ac       float64
cluster                int64
dtype: object

## Create dummy features for categorical variables

**<font color='teal'> Create dummy variables for `state`. Add the dummies back to the dataframe and remove the original column for `state`. </font>**

Hint: you can see an example of how to execute this in Aiden's article on preprocessing [here](https://medium.com/@aiden.dataminer/the-data-science-method-dsm-pre-processing-and-training-data-development-fd2d75182967). 

In [79]:
df1 = pd.concat([df.drop(['state'], axis=1), pd.get_dummies(df['state'])], axis=1)

In [80]:
df1.shape

(157, 61)

In [9]:
df1.describe().T

count         mean          std    min     25%     50%  \
Unnamed: 0         157.0   167.719745    96.460822    1.0    80.0   171.0   
Unnamed: 0.1       157.0   167.719745    96.460822    1.0    80.0   171.0   
summit_elev        157.0  3714.038217  3226.731258  315.0  1270.0  2025.0   
vertical_drop      157.0   851.738854   574.433212  175.0   400.0   700.0   
trams              157.0     0.000000     0.000000    0.0     0.0     0.0   
fastEight          157.0     0.000000     0.000000    0.0     0.0     0.0   
fastSixes          157.0     0.000000     0.000000    0.0     0.0     0.0   
fastQuads          157.0     0.242038     0.592704    0.0     0.0     0.0   
quad               157.0     0.522293     0.703344    0.0     0.0     0.0   
triple             157.0     1.216561     1.257381    0.0     0.0     1.0   
double             157.0     1.757962     1.503847    0.0     1.0     1.0   
surface            157.0     2.222930     1.421442    0.0     1.0     2.0   
total_chairs       157.0     5.961783     2.770924    0.0     4.0     5.0   
Runs               157.0    29.840764    17.499271    0.0    16.0    25.0   
TerrainParks       157.0     1.808917     1.410282    0.0     1.0     2.0   
LongestRun_mi      157.0     1.052229     0.726564    0.0     0.5     1.0   
SkiableTerrain_ac  157.0   230.738854   281.623866    0.0    60.0   125.0   
Snow Making_ac     157.0    84.738854    72.492453    0.0    30.0    60.0   
daysOpenLastYear   157.0    92.165605    40.115833    0.0    75.0   100.0   
yearsOpen          157.0    58.484076    14.732489   22.0    50.0    58.0   
averageSnowfall    157.0   144.006369   110.810389    0.0    50.0   108.0   
AdultWeekday       157.0    50.425766    15.874440   17.0    40.0    49.0   
AdultWeekend       157.0    57.638274    15.314742   20.0    47.0    58.0   
projectedDaysOpen  157.0   109.063694    26.975697   30.0    93.0   107.0   
NightSkiing_ac     157.0    44.471338    48.790967    0.0     0.0    35.0   
cluster            157.0     0.617834     0.772401    0.0     0.0     0.0   
Alaska             157.0     0.012739     0.112504    0.0     0.0     0.0   
Arizona            157.0     0.006369     0.079809    0.0     0.0     0.0   
California         157.0     0.082803     0.276465    0.0     0.0     0.0   
Colorado           157.0     0.025478     0.158075    0.0     0.0     0.0   
Connecticut        157.0     0.031847     0.176155    0.0     0.0     0.0   
Idaho              157.0     0.025478     0.158075    0.0     0.0     0.0   
Illinois           157.0     0.019108     0.137344    0.0     0.0     0.0   
Indiana            157.0     0.012739     0.112504    0.0     0.0     0.0   
Iowa               157.0     0.006369     0.079809    0.0     0.0     0.0   
Maine              157.0     0.031847     0.176155    0.0     0.0     0.0   
Maryland           157.0     0.006369     0.079809    0.0     0.0     0.0   
Massachusetts      157.0     0.019108     0.137344    0.0     0.0     0.0   
Michigan           157.0     0.076433     0.266540    0.0     0.0     0.0   
Minnesota          157.0     0.050955     0.220610    0.0     0.0     0.0   
Missouri           157.0     0.006369     0.079809    0.0     0.0     0.0   
Montana            157.0     0.012739     0.112504    0.0     0.0     0.0   
Nevada             157.0     0.012739     0.112504    0.0     0.0     0.0   
New Hampshire      157.0     0.063694     0.244989    0.0     0.0     0.0   
New Mexico         157.0     0.038217     0.192332    0.0     0.0     0.0   
New York           157.0     0.095541     0.294902    0.0     0.0     0.0   
North Carolina     157.0     0.025478     0.158075    0.0     0.0     0.0   
Ohio               157.0     0.019108     0.137344    0.0     0.0     0.0   
Oregon             157.0     0.019108     0.137344    0.0     0.0     0.0   
Pennsylvania       157.0     0.076433     0.266540    0.0     0.0     0.0   
Rhode Island       157.0     0.006369     0.079809    0.0     0.0     0.0   
South Dakota   

## Standardize the magnitude of numeric features

**<font color='teal'> Using sklearn preprocessing standardize the scale of the features of the dataframe except the name of the resort which we done't need in the dataframe for modeling, so it can be droppped here as well. Also, we want to hold out our response variable(s) so we can have their true values available for model performance review. Let's set `AdultWeekend` to the y variable as our response for scaling and modeling. Later we will go back and consider the `AdultWeekday`, `dayOpenLastYear`, and `projectedDaysOpen`. For now leave them in the development dataframe. </font>**

In [81]:
# first we import the preprocessing package from the sklearn library
from sklearn import preprocessing as pps

# Declare an explanatory variable, called X,and assign it the result of dropping 'Name' and 'AdultWeekend' from the df
X1 = df1.drop(['Name','AdultWeekend'], axis=1)

# Declare a response variable, called y, and assign it the AdultWeekend column of the df 
y1 = df1.AdultWeekend 

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler1 =pps.StandardScaler().fit(X1)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X1_scaled=scaler1.transform(X1) 

## Split into training and testing datasets

**<font color='teal'> Using sklearn model selection import train_test_split, and create a 75/25 split with the y = `AdultWeekend`. We will start by using the adult weekend ticket price as our response variable for modeling.</font>**

In [82]:
# Import the train_test_split function from the sklearn.model_selection utility.  
from sklearn.model_selection import train_test_split

# Get the 1-dimensional flattened array of our response variable y by calling the ravel() function on y
y1 = y1.ravel()

# Call the train_test_split() function with the first two parameters set to X_scaled and y 
# Declare four variables, X_train, X_test, y_train and y_test separated by commas 
X1_train, X1_test, y1_train, y1_test = train_test_split(X1_scaled, y1, test_size=0.25, random_state=1)

In [83]:
type(X1_test)

numpy.ndarray

Here we start the actual modeling work. First let's fit a multiple linear regression model to predict the `AdultWeekend` price.

# Guided Capstone Step 5. Modeling


This is the fifth step in the Data Science Method. In the previous steps you cleaned and prepared the datasets. Now it's time to get into the most exciting part: modeling! In this exercise, you'll build three different models and compare each model's performance. In the end, you'll choose the best model for demonstrating insights to Big Mountain management.



### **The Data Science Method**  


1.   Problem Identification 

2.   Data Wrangling 
  
3.   Exploratory Data Analysis 
 
4.   Pre-processing and Training Data Development

5.   **Modeling**
  * Fit Models with Training Data Set
  * Review Model Outcomes — Iterate over additional models as needed.
  * Identify the Final Model

6.   Documentation
  * Review the Results
  * Present and share your findings - storytelling
  * Finalize Code 
  * Finalize Documentation

## Fit Models with a Training Dataset

**<font color='teal'> Using sklearn, fit the model on your training dataset.</font>**

#### Model 1

In [84]:
#all first model set
from sklearn import linear_model
from sklearn.metrics import explained_variance_score,mean_absolute_error
lm1 = linear_model.LinearRegression()
model1 = lm1.fit(X1_train,y1_train)

**<font color='teal'> Predict on the testing dataset and score the model performance with the y_test set and the y-pred values. The explained variance is a measure of the variation explained by the model. This is also known as the R-squared value. </font>**

Hint: you will have to use the `predict()` method here as it's used in this [DSM article](https://medium.com/@aiden.dataminer/the-data-science-method-dsm-modeling-56b4233cad1b) about modeling. 

In [85]:
# Make a variable called y_pred and assign it the result of calling predict() on our model variable with parameter X_test
y1_pred = lm1.predict(X1_test)

## Review Model Outcomes — Iterate over additional models as needed

In [73]:
# You might want to use the explained_variance_score() and mean_absolute_error() metrics.
# To do so, you will need to import them from sklearn.metrics. 
# You can plug y_test and y_pred into the functions to evaluate the model
from sklearn.metrics import explained_variance_score, mean_absolute_error
variance1=explained_variance_score(y1_test,y1_pred)
absmean1=mean_absolute_error(y1_test,y1_pred)

In [86]:
variance1

-1.180494751176763e+23

In [87]:
absmean1

1566216410978.4543

**<font color='teal'> Print the intercept value from the linear model. </font>**

Hint: our linear regression model `lm` has an attribute `intercept_` for the intercept

In [88]:
Intercept1=lm1.intercept_

In [102]:
Intercept1

5477108021603.002

In [103]:
# print variance and absolute mean errors
print('Variance is: '+str(variance1)+';'+' Absolute mean errors is:'+str(absmean1)+'.')

print('Intercept is:'+str(Intercept1))

Variance is: -1.180494751176763e+23; Absolute mean errors is:1566216410978.4543.
Intercept is:5477108021603.002


**<font color='teal'> The intercept is the mean `AdultWeekend` price for all the resorts given the other characteristics. The addition or subtraction of each of the coefficient values in the regression are numeric adjustments applied to the intercept to provide a particular observation's value for the resulting `AdultWeekend` value. Also, because we took the time to scale our x values in the training data, we can compare each of the coeeficients for the features to determine the feature importances. Print the coefficient values from the linear model and sort in descending order to identify the top ten most important features.</font>** 


Hint: make sure to review the absolute value of the coefficients, because the adjustment may be positive or negative, but what we are looking for is the magnitude of impact on our response variable.

In [90]:
# You might want to make a pandas DataFrame displaying the coefficients for each state like so: 
dfcoeff1 = pd.DataFrame(abs(lm1.coef_), X1.columns, columns=['Coefficient'])

In [29]:
dfcoeff1.sort_values('Coefficient', ascending=False)

Coefficient
trams              1.176860e+14
total_chairs       1.028949e+14
double             5.584354e+13
surface            5.278353e+13
fastSixes          5.235870e+13
Maryland           4.774413e+13
triple             4.669132e+13
quad               2.611782e+13
fastEight          2.343142e+13
fastQuads          2.200936e+13
Rhode Island       1.872047e+13
New York           1.717229e+13
California         1.609872e+13
South Dakota       1.588636e+13
Pennsylvania       1.552076e+13
Michigan           1.552076e+13
Wisconsin          1.491115e+13
New Hampshire      1.426583e+13
Minnesota          1.284625e+13
Vermont            1.205683e+13
New Mexico         1.119960e+13
Maine              1.025759e+13
Connecticut        1.025759e+13
North Carolina     9.204794e+12
Idaho              9.204794e+12
Colorado           9.204794e+12
West Virginia      7.997594e+12
Washington         7.997594e+12
Wyoming            7.997594e+12
Utah               7.997594e+12
Ohio               7.997594e+12
Oregon             7.997594e+12
Massachusetts      7.997594e+12
Illinois           7.997594e+12
Virginia           6.551175e+12
Nevada             6.551175e+12
Montana            6.551175e+12
Indiana            6.551175e+12
Alaska             6.551175e+12
Tennessee          4.647300e+12
Missouri           4.647300e+12
Iowa               4.647300e+12
Arizona            4.647300e+12
Unnamed: 0.1       1.064053e+01
AdultWeekday       9.917969e+00
Unnamed: 0         9.864869e+00
summit_elev        4.193530e+00
vertical_drop      1.699386e+00
SkiableTerrain_ac  1.320312e+00
averageSnowfall    9.765625e-01
LongestRun_mi      9.409180e-01
NightSkiing_ac     8.906250e-01
yearsOpen          8.730469e-01
Runs               6.718750e-01
TerrainParks       4.296875e-01
projectedDaysOpen  2.939453e-01
daysOpenLastYear   2.929688e-01
Snow Making_ac     9.375000e-02
cluster            2.148438e-02

**<font color='teal'>You should see that the top ten important features are different states. However, the state is not something the managers at the Big Mountain Resort can do anything about. Given that we care more about actionable traits associated with ticket pricing, rebuild the model without the state features and compare the results. </font>**

Hint: Try to construct another model using exactly the steps we followed above. 

#### Model 2

In [98]:
# Declare an explanatory variable, called X2,and assign it the result of dropping 'Name', 'state' and 'AdultWeekend' from the df
X2= df.drop(['Name','state','AdultWeekend'], axis=1)

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler2 =pps.StandardScaler().fit(X2)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X2_scaled=scaler2.transform(X2) 

# New training and test dataset for X2

X2_train, X2_test, y2_train, y2_test = train_test_split(X2_scaled, y1, test_size=0.25, random_state=1)

# running linear regression on model 2
lm2 = linear_model.LinearRegression()
model2 = lm2.fit(X2_train,y2_train)

#compute predicted y2 from model2
y2_pred = lm2.predict(X2_test)

#check variance and absolute mean errors for model2
variance2 = explained_variance_score(y2_test,y2_pred)
absmean2 = mean_absolute_error(y2_test,y2_pred)

# print variance and absolute mean errors
print('Variance is: '+str(variance2)+';'+' Absolute mean errors is:'+str(absmean2)+'.')

# get intercept and print it
Intercept2=lm2.intercept_

print('Intercept is:'+str(Intercept2))

# Make a pandas DataFrame displaying the coefficients for each state like so: 
dfcoeff2 = pd.DataFrame(abs(lm2.coef_), X2.columns, columns=['Coefficient'])

dfcoeff2.sort_values('Coefficient', ascending=False)

Variance is: 0.7605920580354469; Absolute mean errors is:6.219899911342127.
Intercept is:57.89792913075968


Coefficient
AdultWeekday       1.043106e+01
vertical_drop      2.529861e+00
cluster            2.513310e+00
summit_elev        2.346390e+00
SkiableTerrain_ac  1.303819e+00
averageSnowfall    1.264760e+00
quad               1.236396e+00
triple             1.157658e+00
TerrainParks       9.177650e-01
NightSkiing_ac     7.790109e-01
total_chairs       6.580974e-01
Runs               6.510294e-01
yearsOpen          5.898333e-01
projectedDaysOpen  4.080068e-01
fastQuads          3.562106e-01
LongestRun_mi      2.162116e-01
Unnamed: 0         1.991588e-01
Unnamed: 0.1       1.991588e-01
daysOpenLastYear   1.846948e-01
double             1.603112e-01
surface            3.480948e-02
Snow Making_ac     2.194356e-02
fastSixes          1.332268e-15
fastEight          2.220446e-16
trams              0.000000e+00

**<font color='teal'> When reviewing our new model coefficients, we see `summit_elev` is now in the number two spot. This is also difficult to change from a management prespective and highly correlated with `base_elev` and `vertical_drop`.  This time, rebuild the model without the state features and without the `summit_elev` and without `base_elev`and compare the results. </font>**

#### Model 3

In [100]:
# Declare an explanatory variable, called X2,and assign it the result of dropping 'Name', 'state' , 'summit_elev', and 'AdultWeekend' from the df
# 'base_elev' has been dropped in step 3

X3= df.drop(['Name','state','summit_elev','AdultWeekend'], axis=1)

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler3 =pps.StandardScaler().fit(X3)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X3_scaled=scaler3.transform(X3) 

# New training and test dataset for X3

X3_train, X3_test, y3_train, y3_test = train_test_split(X3_scaled, y1, test_size=0.25, random_state=1)

# running linear regression on model 3
lm3 = linear_model.LinearRegression()
model3 = lm3.fit(X3_train,y3_train)

#compute predicted y2 from model3
y3_pred = lm3.predict(X3_test)

#check variance and absolute mean errors for model3
variance3 = explained_variance_score(y3_test,y3_pred)
absmean3 = mean_absolute_error(y3_test,y3_pred)

# print variance and absolute mean errors
print('Variance is: '+str(variance3)+';'+' Absolute mean errors is:'+str(absmean3)+'.')

# get intercept and print it
Intercept3=lm3.intercept_

print('Intercept is:'+str(Intercept3))

# Make a pandas DataFrame displaying the coefficients for each state like so: 
dfcoeff3 = pd.DataFrame(abs(lm3.coef_), X3.columns, columns=['Coefficient'])

dfcoeff3.sort_values('Coefficient', ascending=False)

Variance is: 0.7443566899941869; Absolute mean errors is:6.562525377322362.
Intercept is:57.89973683311378


Coefficient
AdultWeekday       1.031098e+01
vertical_drop      2.330672e+00
cluster            1.835771e+00
SkiableTerrain_ac  1.805514e+00
averageSnowfall    1.590582e+00
quad               1.127434e+00
TerrainParks       1.015884e+00
NightSkiing_ac     9.047875e-01
triple             8.774807e-01
total_chairs       7.341915e-01
projectedDaysOpen  5.224298e-01
fastQuads          5.206435e-01
Runs               4.990371e-01
Unnamed: 0         3.902046e-01
Unnamed: 0.1       3.902046e-01
yearsOpen          3.828676e-01
surface            3.429185e-01
Snow Making_ac     2.485219e-01
LongestRun_mi      1.443033e-01
daysOpenLastYear   5.222386e-02
double             2.710579e-02
fastEight          1.776357e-15
fastSixes          1.332268e-15
trams              4.440892e-16

## Identify the Final Model

**<font color='teal'> Review the model performances in the table below and choose the best model for proving insights to Big Mountain management about what features are driving ski resort lift ticket prices. Type your choice in the final markdown cell — you will discuss this selection more in the next step of the guided casptone. </font>**

| Model | Explained Variance| Mean Absolute Error|Features Dropped|
| --- | --- | --- | --- |
| Model 1. | -1.180494751176763e+23|1566216410978.4543  ||
| Model 2. | 0.7605920580354469| 6.219899911342127|'state'|
| Model 3. | 0.7443566899941869 |6.562525377322362 |'state','summit_elev'|

Model Selection:

# Model 2 and Model 3 are close. Would recommend to take model 3.

Model 2 and Model 3 are close. Would recommend to take model 3.